numpy/core/_utils.py

```diff
@@ -0,0 +1,15 @@
+import warnings
+
+
+def _raise_warning(attr: str) -> None:
+
+    warnings.warn(
+        f"`numpy.core` has been made officially private and renamed to "
+        "`numpy._core`. Accessing `_core` directly is discouraged, as "
+        "members of `_core` can change without any warning at any time. "
+        "You should use a public module instead that exports the attribute "
+        "in question. If you still would like to access an attribute "
+        f"from it, please use `np._core.{attr}`.",
```

Comment:
You refer to the module as `numpy` above, use either that or `np` but not both.

Also looking at the text of other warnings, I don't think it's correct to include the backticks around paths. These are going to be displayed in a terminal so the markdown formatting isn't helpful.

Reply:
I agree - I updated the warning message.